f5lbaasdriver/v2/bigip/service_builder.py

```diff
@@ -277,20 +281,26 @@ def _populate_member_network(self, context, member, network):
         member['vxlan_vteps'] = []
         member['gre_vteps'] = []
 
-        if 'provider:network_type' in network:
-            net_type = network['provider:network_type']
-            if net_type == 'vxlan':
-                if 'binding:host_id' in member['port']:
-                    host = member['port']['binding:host_id']
-                    member['vxlan_vteps'] = self._get_endpoints(
-                        context, 'vxlan', host)
-            if net_type == 'gre':
-                if 'binding:host_id' in member['port']:
-                    host = member['port']['binding:host_id']
-                    member['gre_vteps'] = self._get_endpoints(
-                        context, 'gre', host)
-        if 'provider:network_type' not in network:
-            network['provider:network_type'] = 'undefined'
+        agent_config = {}
+        segment_data = self.disconnected_service.get_network_segment(
+            context, agent_config, network)
+        if segment_data:
```

Comment:
Lines 91-101 in build() could be moved to a helper function and called here in place of lines 284-291.  Future-proof the code a bit if/when we change that helper for how network attributes are remapped.